nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/kinesis/ITPutKinesis.java

```diff
@@ -0,0 +1,475 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws.kinesis;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.nifi.processors.aws.kinesis.stream.PutKinesis;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * This test contains both unit and integration test (integration tests are ignored by default)
```

Comment:
Could we add a comment about what kind of Kinesis Stream setup is required to run these tests?  I was able to run all of the integration tests, but it would have helped if I knew how many shards were required.  I started with just one shard, because I'm cheap, and that wasn't enough to get through the tests that exceed the 1 MB/sec provisioned throughput.  Testing with various shard sizes suggests the tests are logically OK, the only issue seems to be individual and collective provisioned throughput.  Finally, I forked over the extra $0.075 for five shards and the unit tests worked fine together.  I agree with testing the 1 MB buffer max handling, I just didn't know how to set it up right the first time.

What was your setup like?